sdk/monitor/opentelemetry-exporter-azuremonitor/opentelemetry/exporter/azuremonitor/export/trace/__init__.py

```diff
@@ -0,0 +1,180 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+import json
+import logging
+from typing import Sequence
+from urllib.parse import urlparse
+
+from opentelemetry.sdk.trace.export import SpanExporter, SpanExportResult
+from opentelemetry.sdk.util import ns_to_iso_str
+from opentelemetry.trace import Span, SpanKind
+from opentelemetry.trace.status import StatusCanonicalCode
+
+from opentelemetry.exporter.azuremonitor import utils
+from opentelemetry.exporter.azuremonitor._generated.models import (
+    MonitorBase,
+    RemoteDependencyData,
+    RequestData,
+    TelemetryItem
+)
+from opentelemetry.exporter.azuremonitor.export import (
+    BaseExporter,
+    ExportResult,
+    get_trace_export_result,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class AzureMonitorSpanExporter(BaseExporter, SpanExporter):
+    """Azure Monitor span exporter for OpenTelemetry.
+
+    Args:
+        options: :doc:`export.options` to allow configuration for the exporter
+    """
+
+    def __init__(self, **options):
+        super().__init__(**options)
+        self.add_telemetry_processor(indicate_processed_by_metric_extractors)
+
+    def export(self, spans: Sequence[Span]) -> SpanExportResult:
+        envelopes = list(map(self._span_to_envelope, spans))
+        envelopes = list(
+            map(
+                lambda x: x,
+                self._apply_telemetry_processors(envelopes),
+            )
+        )
+        try:
+            result = self._transmit(envelopes)
+            if result == ExportResult.FAILED_RETRYABLE:
+                self.storage.put(envelopes, result)
+            if result == ExportResult.SUCCESS:
+                # Try to send any cached events
+                self._transmit_from_storage()
+            return get_trace_export_result(result)
+        except Exception:  # pylint: disable=broad-except
+            logger.exception("Exception occurred while exporting the data.")
+            return get_trace_export_result(ExportResult.FAILED_NOT_RETRYABLE)
+
+    # pylint: disable=too-many-statements
+    # pylint: disable=too-many-branches
+    def _span_to_envelope(self, span: Span) -> TelemetryItem:
+        if not span:
+            return None
+        envelope = convert_span_to_envelope(span)
+        envelope.instrumentation_key = self.options.instrumentation_key
+        return envelope
+
+
+# pylint: disable=too-many-statements
+# pylint: disable=too-many-branches
+def convert_span_to_envelope(span: Span) -> TelemetryItem:
+    if not span:
+        return None
+    envelope = TelemetryItem(
+        name="",
+        instrumentation_key="",
+        tags=dict(utils.azure_monitor_context),
+        time=ns_to_iso_str(span.start_time),
+    )
+    envelope.tags["ai.operation.id"] = "{:032x}".format(span.context.trace_id)
+    parent = span.parent
+    if isinstance(parent, Span):
+        parent = parent.context
```

Comment:
https://docs.python.org/3/glossary.html#term-eafp

```suggestion
    try:
        parent = parent.context
    except AttributeError:
        pass
```

Reply:
Oh good catch. This was actually because prior to [this change](https://github.com/open-telemetry/opentelemetry-python/pull/1146/files) in Opentelemetry, the `parent` could actually be a `Span` or `SpanContext`, depending on whether it was the root span. But now after that change, `parent` is always the span's parent's `SpanContext`.

Reply:
Removed unnecessary check